### PR TITLE
Fixes clippy warnings

### LINKED ISSUE #239: Fix clippy warnings

### PR DESCRIPTION
**Pull request by sanpii**
Fixes #239 & #254

This PR is incomplete, there is still 4 warnings:
<details>

```
    Checking relm-derive v0.21.0 (/home/sanpi/sources/relm/relm-derive)                                                                                          [5/74]
warning: large size difference between variants
  --> src/gen/parser.rs:88:5
   |
88 |     Return(Expr, Expr),
   |     ^^^^^^^^^^^^^^^^^^ this variant is 560 bytes
   |
   = note: `#[warn(clippy::large_enum_variant)]` on by default
note: and the second-largest variant is 280 bytes:
  --> src/gen/parser.rs:87:5
   |
87 |     CallReturn(Expr),
   |     ^^^^^^^^^^^^^^^^
help: consider boxing the large fields to reduce the total size of the enum
  --> src/gen/parser.rs:88:5
   |
88 |     Return(Expr, Expr),
   |     ^^^^^^^^^^^^^^^^^^
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#large_enum_variant

warning: this function has too many arguments (8/7)
   --> src/gen/parser.rs:137:5
    |
137 | /     fn new_gtk(widget: GtkWidget, typ: Path, init_parameters: Vec<Expr>, children: Vec<Widget>,
138 | |         properties: HashMap<Ident, Expr>, child_properties: ChildProperties, child_events: ChildEvents,
139 | |         nested_views: HashMap<Ident, Widget>) -> Self
    | |_____________________________________________________^
    |
    = note: `#[warn(clippy::too_many_arguments)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: this function has too many arguments (8/7)
   --> src/gen/parser.rs:160:5
    |
160 | /     fn new_relm(widget: RelmWidget, typ: Path, init_parameters: Vec<Expr>, children: Vec<Widget>,
161 | |         properties: HashMap<Ident, Expr>, child_properties: ChildProperties, child_events: ChildEvents,
162 | |         nested_views: HashMap<Ident, Widget>) -> Self
    | |_____________________________________________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#too_many_arguments

warning: very complex type used. Consider factoring parts into `type` definitions
  --> src/gen/generator.rs:65:70
   |
65 | ...driver: &mut Driver) -> (TokenStream, HashMap<Ident, Path>, HashMap<Ident, Path>, HashSet<Ident>, TokenStream) {
   |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(clippy::type_complexity)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#type_complexity

warning: 4 warnings emitted

    Finished dev [unoptimized + debuginfo] target(s) in 0.50s
```
</details>

I don’t know how fix that.

### CODE REVIEW

**Comment by antoyo** [comment]
For the first warning, I guess it would be safer to `Box` the `Expr` in the variant `Return` to avoid a stack overflow in some cases.

For the functions with too many parameters, I'd just ignore them, as these are initializing structs with many fields anyway, so adding `#[allow(clippy::too_many_arguments)]` should do.

For the last error (src/gen/generator.rs:65:70), just make the return type be a `struct` instead of a tuple.

Also, it would be nice to run `clippy` [somewhere in the CI](https://github.com/antoyo/relm/blob/master/.github/workflows/CI.yml#L34).

Thanks for looking into this!

**Comment by sanpii** [comment]
> For the first warning, I guess it would be safer to `Box` the `Expr` in the variant `Return` to avoid a stack overflow in some cases.

Done, but destructuring `Box` isn’t easy as a tuple.

> For the functions with too many parameters, I'd just ignore them, as these are initializing structs with many fields anyway, so adding `#[allow(clippy::too_many_arguments)]` should do.

Ok, done.

> For the last error (src/gen/generator.rs:65:70), just make the return type be a `struct` instead of a tuple.

I just created a type to limit API breaking. Why the `gen` function is public? Public for crate is not enough? If you are ok, I can write another PR to restrict function visibility and change its return struct.

> Also, it would be nice to run `clippy` [somewhere in the CI](https://github.com/antoyo/relm/blob/master/.github/workflows/CI.yml#L34).

Done.

**Comment by antoyo** [comment]
Yep, the visibility can be `crate`.

File: src/core/mod.rs [hunk 1]
```diff
@@ -266,9 +268,14 @@ impl<MSG> Drop for EventStream<MSG> {
     }
 }
 
+impl<MSG> Default for EventStream<MSG> {
```
Comment (antoyo) [inline src/core/mod.rs:271]: I don't really like this clippy lint requiring you to have a `Default` implementation.
Can you just disable it globally, please?

File: relm-derive/src/gen/mod.rs [hunk 2]
```diff
@@ -28,7 +28,7 @@
 pub(crate) mod parser;
 
 mod adder;
-mod gen;
+mod generator;
```
Comment (antoyo) [inline relm-derive/src/gen/mod.rs:31]: Was there a specific reason to rename the module?

**Review by antoyo — CHANGES_REQUESTED** [review]
Sorry, I missed a few things in the first review.

File: relm-derive/src/gen/mod.rs (hunk 2 above)
Comment (sanpii) [reply relm-derive/src/gen/mod.rs:31]: https://rust-lang.github.io/rust-clippy/master/index.html#module_inception

Comment (antoyo) [reply relm-derive/src/gen/mod.rs:31]: :+1: I guess we can just use `generator` until I find better names ;) .

**Review by antoyo — APPROVED** [review]
Thanks!

I guess we could run `clippy` on the examples, but that probably requires fixing the code generation, so let's stop here for this PR.

Ready to merge?

**Comment by sanpii** [comment]
> Ready to merge?

Yes, it’s ok for me.

**Pull request merged**